### PR TITLE
feat(page-user): add user page with general settings

### PR DESCRIPTION
**Pull request by RemiBonnet**
# What does this PR do?

[> Link to the JIRA ticket
](https://qovery.atlassian.net/jira/software/projects/FRT/boards/23?selectedIssue=FRT-718)

Capture d’écran 2023-08-17 à 14 26 37

---

## PR Checklist

- [ ] This PR introduces breaking change(s) and has been labeled as such
- [ ] This PR introduces new store changes
- [x] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`
- [x] I made sure the code is type safe (no any)

### CODE REVIEW

**Comment by acarranoqovery** [comment]
[Qovery Preview](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery  preview 28c47145-c8e7-4b9d-8d9e-c65c95b48425 | [staging](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/28c47145-c8e7-4b9d-8d9e-c65c95b48425/services/general) |
| /qovery  preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> Truth can only be found in one place: the code

File: apps/console/src/app/app.tsx [hunk 1]
```diff
@@ -92,37 +92,32 @@ export function App() {
...
-        )}
-        {ROUTER.map(
-          (route) =>
-            route.layout && (
-              <Route
-                key={route.path}
-                path={route.path}
-                element={
-                  !route.protected ? (
+        {ROUTER.map((route) =>
```
Comment (RemiBonnet) [inline apps/console/src/app/app.tsx:95]: Fixing weird double map

**Comment by acarranoqovery** [comment]
[Qovery Preview](https://www.qovery.com)
---

A preview environment was automatically created via Qovery.
Click on the link below to follow its deployment and use it.
:point_right: [\[PR\] staging - feat(page-user): add user page with general settings - 2023-08-18T07:43:30Z](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/1d071010-f334-42a0-ba07-5d79539d62e6/services/general)

 Another comment will be posted when deployments are terminated

**Comment by acarranoqovery** [comment]
[Qovery Preview](https://www.qovery.com)
---

Your preview environment has been successfully deployed !
Click on the link below to open your service:
:point_right: [console](https://760-console-staging.zc531a994.rustrocks.cloud)

File: libs/pages/user/src/lib/feature/page-user-general-feature/page-user-general-feature.tsx [hunk 2]
```diff
@@ -23,7 +23,7 @@ export function PageUserGeneralFeature() {
     defaultValues: {
       firstName: user.first_name,
       lastName: user.last_name,
-      email: user.user_email,
+      email: user.user_email !== '' ? user.user_email : userToken.email!,
```
Comment (ctjhoa) [inline libs/pages/user/src/lib/feature/page-user-general-feature/page-user-general-feature.tsx:26]: This doesn't work for me.
This does:
`email: user.user_email ?? userToken.email`

**Review by ctjhoa — CHANGES_REQUESTED** [review]

File: libs/pages/user/src/lib/feature/page-user-general-feature/page-user-general-feature.tsx (hunk 2 above)
Comment (RemiBonnet) [reply libs/pages/user/src/lib/feature/page-user-general-feature/page-user-general-feature.tsx:26]: Thanks, I couldn't test with my account

**Review by ctjhoa — APPROVED** [review]

**Pull request merged**